code/procs/jobprocs.dm

```diff
@@ -309,6 +309,85 @@
 		picks = FindOccupationCandidates(staff,JOB.name,3)
 	return picks
 
+/proc/equip_job_items(var/datum/job/JOB, var/mob/living/carbon/human/H)
+	// Backpack and contents
+	if (JOB.slot_back && length(JOB.slot_back) > 1)
+		H.equip_new_if_possible(weighted_pick(JOB.slot_back), H.slot_back)
+	else if (length(JOB.slot_back))
+		H.equip_new_if_possible(JOB.slot_back[1], H.slot_back)
+	if (JOB.slot_back && length(JOB.items_in_backpack))
+		for (var/X in JOB.items_in_backpack)
+			if(ispath(X))
+				H.equip_new_if_possible(X, H.slot_in_backpack)
+	// Belt and contents
+	if (JOB.slot_belt && length(JOB.slot_belt) > 1)
+		H.equip_new_if_possible(weighted_pick(JOB.slot_belt), H.slot_belt)
+	else if (length(JOB.slot_belt))
+		H.equip_new_if_possible(JOB.slot_belt[1], H.slot_belt)
+	if (JOB.slot_belt && length(JOB.items_in_belt) && istype(H.belt, /obj/item/storage))
+		for (var/X in JOB.items_in_belt)
+			if(ispath(X))
+				H.equip_new_if_possible(X, H.slot_in_belt)
+	// Jumpsuit
+	if (JOB.slot_jump && length(JOB.slot_jump) > 1)
+		H.equip_new_if_possible(weighted_pick(JOB.slot_jump), H.slot_w_uniform)
+	else if (length(JOB.slot_jump))
+		H.equip_new_if_possible(JOB.slot_jump[1], H.slot_w_uniform)
+	// Footwear
+	if (JOB.slot_foot && length(JOB.slot_foot) > 1)
+		H.equip_new_if_possible(weighted_pick(JOB.slot_foot), H.slot_shoes)
+	else if (length(JOB.slot_foot))
+		H.equip_new_if_possible(JOB.slot_foot[1], H.slot_shoes)
+	// Suit
+	if (JOB.slot_suit && length(JOB.slot_suit) > 1)
+		H.equip_new_if_possible(weighted_pick(JOB.slot_suit), H.slot_wear_suit)
+	else if (length(JOB.slot_suit))
+		H.equip_new_if_possible(JOB.slot_suit[1], H.slot_wear_suit)
+	// Ears
+	if (JOB.slot_ears)
+		if (!(H.traitHolder && H.traitHolder.hasTrait("allears") && ispath(JOB.slot_ears,/obj/item/device/radio/headset)))
+			H.equip_new_if_possible(JOB.slot_ears[1], H.slot_ears)
```

Comment:
I think you need to check length of JOB.slot_ears at 347 like with the other slots WRT the map compiles runtiming, They're complaining about an index on 349.